src/sandstorm/run-bundle.c++

```diff
@@ -1951,10 +1949,23 @@ private:
 
     pid_t updaterPid = startUpdater(config, fdBundle, false);
 
-    pid_t sandstormPid = fork();
-    if (sandstormPid == 0) {
-      runServerMonitor(config, fdBundle);
-      KJ_UNREACHABLE;
+    // Start the server monitor.
+    pid_t sandstormPid;
+    {
+      int cloneFlags = CLONE_PARENT_SETTID | SIGCHLD;
+      if(!runningAsRoot) {
+        cloneFlags |= CLONE_NEWUSER|CLONE_NEWPID;
+        unsharedUidNamespace = true;
+      }
+
+      uid_t uid = getuid();
+      gid_t gid = getgid();
+
+      KJ_SYSCALL(syscall(SYS_clone, cloneFlags, nullptr, &sandstormPid, 0));
+      if(sandstormPid == 0) {
+        runServerMonitor(config, fdBundle, uid, gid);
+        KJ_UNREACHABLE;
+      }
```

Comment:
I think `sandstormPid` is left uninitialized in the child process (which might explain the hang in CI?). clone() actually returns the PID just like fork() does, you don't need to use the CLONE_PARENT_SETTID feature (I'm not totally sure why that feature exists...).

```suggestion
    pid_t sandstormPid;
    {
      int cloneFlags = SIGCHLD;
      if(!runningAsRoot) {
        cloneFlags |= CLONE_NEWUSER | CLONE_NEWPID;
        unsharedUidNamespace = true;
      }

      uid_t uid = getuid();
      gid_t gid = getgid();

      KJ_SYSCALL(sandstormPid = syscall(SYS_clone, cloneFlags, nullptr, nullptr, 0));
      if(sandstormPid == 0) {
        runServerMonitor(config, fdBundle, uid, gid);
        KJ_UNREACHABLE;
      }
```